### PR TITLE
add jenkinsfile

### PR DESCRIPTION
**Pull request by BrandonArp**
Moving to our own build infrastructure. With the loss of Travis, I was finally incentivized to setup a Jenkins instance. This should let us build more advanced scenarios as well as use more resources in our build (and get fancy plugins to report). Project admins should contact me to setup an account on Jenkins to control the builds.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        script {
+          target = "install"
+        }
+      }
+    }
+    stage('Setup release') {
+      when { buildingTag(); not { changeRequest() }  }
+      steps {
+        script {
+          target = "install deploy -P release  --settings settings.xml"
```
Comment (vjkoskela) [inline Jenkinsfile:28]: Shouldn't this have `-P rpm` as well?

Comment (vjkoskela) [reply Jenkinsfile:28]: There are some other useful flags we've used in the past as well:

* `-U` - Update all package versions (avoids negative caching)
* `-B` - Batch mode short circuits any prompts for user input
* `-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn` - Reduces logging

Comment (vjkoskela) [reply Jenkinsfile:28]: Ref:
https://github.com/ArpNetworking/metrics-aggregator-daemon/blob/master/.github/workflows/build.yml#L32

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    stage('Init') {
+      steps {
+        checkout scm
+      }
+    }
+    stage('Setup build') {
+      when { not { buildingTag() } }
+      steps {
+        script {
+          target = "install"
```
Comment (vjkoskela) [inline Jenkinsfile:20]: Missing Maven arguments? (see below)

**Review by vjkoskela — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (BrandonArp) [reply Jenkinsfile:28]: Updated. What is the stuff from today about the pooled connections and keep-alives?

Comment (vjkoskela) [reply Jenkinsfile:28]: It wasn't a problem on Travis, and I would be surprised it was a problem egressing your build network. I believe you do pay a penalty wrt performance with the settings that were prescribed for Azure. I'm okay going back to baseline here.

**Review by vjkoskela — APPROVED** [review]

**Pull request merged**